### PR TITLE
Add service-fabrik-deployment-hooks job and package

### PR DESCRIPTION
**Pull request by amitmalav**
* Added separate job for deployment hooks
* Added separate package for deployment hooks

### CODE REVIEW

File: jobs/service-fabrik-deployment-hooks/spec [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  config/settings.yml.erb: config/settings.yml
+  config/eventlog-config-hook.yml.erb: config/eventlog-config-hook.yml
+
+provides:
+- name: deployment_hooks
+  type: deployment_hooks
+  properties:
+  - username
+  - password
+  - skip_ssl_validation
```
Comment (jagadish-kb) [inline jobs/service-fabrik-deployment-hooks/spec]: dont provide what you feel will not be used by other jobs. for ex, all these log_level, enable_circuit_breaker etc.. are internal to this job, why bother providing to other jobs which is not going to be used. So provide only the only the ones thats required.

File: jobs/service-fabrik-deployment-hooks/templates/config/eventlog-config-hook.yml.erb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+---
```
Comment (jagadish-kb) [inline jobs/service-fabrik-deployment-hooks/templates/config/eventlog-config-hook.yml.erb]: not required to event log this, but this is my opinion. Please get it verified.

File: jobs/service-fabrik-report/templates/config/settings.yml.erb [hunk 3]
```diff
@@ -59,6 +59,16 @@ production:
     <% end.else do %>
     protocol: 'http'
     <% end %>
+
```
Comment (jagadish-kb) [inline jobs/service-fabrik-report/templates/config/settings.yml.erb]: why does report need this config?

File: jobs/service-fabrik-scheduler/templates/config/settings.yml.erb [hunk 4]
```diff
@@ -70,6 +70,15 @@ production:
     protocol: 'http'
     <% end %>
 
+  #############################
+  # DEPLOYMENT HOOKS SETTINGS #
+  #############################
+  deployment_hooks:
```
Comment (jagadish-kb) [inline jobs/service-fabrik-scheduler/templates/config/settings.yml.erb]: why does scheduler need this config? Hooks IMO, should be sth that only broker must access its an extension of broker but just to sandbox this is being done. So no other external entities must ever communicate with hooks server.

File: jobs/service-fabrik-deployment-hooks/spec (hunk 1 above)
Comment (amitmalav) [reply jobs/service-fabrik-deployment-hooks/spec]: Okay. Was not aware of this feature. I though all the properties needs to be in provided array. Will make the correction

File: jobs/service-fabrik-report/templates/config/settings.yml.erb (hunk 3 above)
Comment (amitmalav) [reply jobs/service-fabrik-report/templates/config/settings.yml.erb]: report process does not need this setting per se. But currently `deploymentHookClient ` exports client instance which is loaded by `DirectorManager`. And `DirectorManager` is loaded by `index.js` of lib. So it becomes mandatory for processes using broker module to have deployment_hooks settings :(
So in current situation we have to add almost all the configs from broker to report as well as scheduler.
One more reason to modularise our source code.

File: jobs/service-fabrik-scheduler/templates/config/settings.yml.erb (hunk 4 above)
Comment (amitmalav) [reply jobs/service-fabrik-scheduler/templates/config/settings.yml.erb]: Same as above

File: jobs/service-fabrik-report/templates/config/settings.yml.erb (hunk 3 above)
Comment (jagadish-kb) [reply jobs/service-fabrik-report/templates/config/settings.yml.erb]: in current setup as well you can avoid it by exporting `DeploymentHookClient` class and not its instance and DirectorManager lazily creating an instance of it when required. That should also address the above isssue?

File: jobs/service-fabrik-deployment-hooks/spec (hunk 1 above)
Comment (amitmalav) [reply jobs/service-fabrik-deployment-hooks/spec]: Done

File: jobs/service-fabrik-report/templates/config/settings.yml.erb (hunk 3 above)
Comment (amitmalav) [reply jobs/service-fabrik-report/templates/config/settings.yml.erb]: Done!

**Pull request merged**